### ISSUE TITLE
feat(StatusQ/SubmodelProxyModel): Generalize `SubmodelProxyModel` to be used not only for submodels

### ISSUE THREAD

**Issue by micieslak** [status-im/status-desktop#14392]
## Description

The natural continuation of https://github.com/status-im/status-desktop/issues/14390 is to make `SubmodelProxyModel` not limited to be used with models containing submodels.

As a result it would be a general purpose proxy model decorating source model with computed roles using arbitrary user-provided qml component:

```
ObjectProxyModel {
    id: submodelProxyModel

    sourceModel: tokensModel
    
    delegate: SortFilterProxyModel {
        id: delegateRoot
        
        readonly property alias balancesFilteredRole: delegateRoot
        readonly property alias totalBalanceRole: aggregator.value
        readonly property string imageUrlRole: getIcon(symbol)

        sourceModel: balances

        filters: ValueFilter {
            roleName: "address"
            value: selectedAddress
        }
        
        SumAggregator {
            id: aggregator

            model: balances
            roleName: "balance"
        }
    }
}
```

It would add 3 new roles to the model: `balancesFiltered`, `totalBalance` and `imageUrl`. Roles can depend on submodels or any other source model's role.

The difficulty here is to provide model's roles for given row in an efficient way. It's not clear if it should/could be done via context properties or in other way - up to further investigation. It's also not clear if they should be pre-declared as in `FastExpression*` components for SFPM.

Depends on: https://github.com/status-im/status-desktop/issues/14390

**Comment by micieslak** [comment]
Duplicate of https://github.com/status-im/status-desktop/issues/14893
